kubernetes/samples/scripts/domain-lifecycle/rollCluster.sh

```diff
@@ -96,10 +96,12 @@ initialize() {
 
 initialize
 
-# Get the domain in json format
-domainJson=$(${kubernetesCli} get domain ${domainUid} -n ${domainNamespace} -o json --ignore-not-found)
-if [ -z "${domainJson}" ]; then
-  printError "Unable to get domain resource for domain '${domainUid}' in namespace '${domainNamespace}'. Please make sure the 'domain_uid' and 'namespace' specified by the '-d' and '-n' arguments are correct. Exiting."
+# Get the domain in json format. Changed made on 08/15/2022
+#domainJson=$(${kubernetesCli} get domain ${domainUid} -n ${domainNamespace} -o json --ignore-not-found)
+clusterJson=$(${kubernetesCli} get cluster ${clusterName} -n ${domainNamespace} -o json --ignore-not-found)
+printInfo "clusterJson content before changes: ${clusterJson}"
+if [ -z "${clusterJson}" ]; then
+  printError "Unable to get cluster resource for cluster '${clusterName}' in namespace '${domainNamespace}'. Please make sure the 'clusterName and 'namespace' specified by the '-d' and '-n' arguments are correct. Exiting."
```

Comment:
The script is assuming that cluster resource will always exist and it's going to fail if the cluster resource doesn't exist. My understanding is that cluster resource may not exist for the default behavior. @rjeberhard The expected behavior of when the cluster resource doesn't exist needs to be decided.

Reply:
I just discussed this with QA folks in today's weekly team meeting, we feel that we should keep the original logic in the scripts - fail the test if there is no cluster exists. The reason is that for this test suite, we are testing "cluster" related operations, like restart, rollingrestart, scale, etc. an existing cluster is a must for the tests to pass and test suite is responsible to create it and then let the script now. If we let the script to create the cluster if it's missing for any reason, it may confuse the test suite and make a mess later.

Reply:
Anil, Robert and I discussed and we've decided that the cluster-related scripts should fail with a helpful error message if the Cluster resource doesn't exist. The scripts and the operator will not create the missing Cluster resource. Cluster resources will only be automatically created in the narrow scope of the v8 -> v9 schema conversion webhook.

Reply:
Changed error msg to "Unable to get cluster resource for cluster '${clusterName}' in namespace '${domainNamespace}'. Please make sure that a Cluster exists for cluster '${clusterName}' and that this Cluster is referenced by the Domain."